src/Azure/Orleans.Clustering.AzureStorage/OrleansSiloInstanceManager.cs

```diff
@@ -218,14 +211,8 @@ private async Task DeleteEntriesBatch(List<Tuple<SiloInstanceTableEntry, string>
         {
             string rowKey = SiloInstanceTableEntry.ConstructRowKey(siloAddress);
 
-            string filterOnPartitionKey = TableQuery.GenerateFilterCondition(nameof(SiloInstanceTableEntry.PartitionKey), QueryComparisons.Equal,
-                    this.DeploymentId);
-            string filterOnRowKey1 = TableQuery.GenerateFilterCondition(nameof(SiloInstanceTableEntry.RowKey), QueryComparisons.Equal,
-                rowKey);
-            string filterOnRowKey2 = TableQuery.GenerateFilterCondition(nameof(SiloInstanceTableEntry.RowKey), QueryComparisons.Equal, SiloInstanceTableEntry.TABLE_VERSION_ROW);
-            string query = TableQuery.CombineFilters(filterOnPartitionKey, TableOperators.And, TableQuery.CombineFilters(filterOnRowKey1, TableOperators.Or, filterOnRowKey2));
-
-            var queryResults = await storage.ReadTableEntriesAndEtagsAsync(query);
+            var filter = TableClient.CreateQueryFilter($"(PartitionKey eq {DeploymentId}) and ((RowKey eq {rowKey}) or (RowKey eq {SiloInstanceTableEntry.TABLE_VERSION_ROW}))");
```

Comment:
`TableOdataFilter.Create`

Reply:
TableOdataFilter is internal, so I am using TableClient instead, which calls into TableOdataFilter

Reply:
Oops - you're correct :)